### DISCUSSION TITLE
errore in Configuration.yaml

### DISCUSSION THREAD

**Discussion by gericom90** [indomus/forum#406] [HUB: Home Assistant]
Buona giornata, sto installando H.A Core su docker.
Home Assistant si apre in modalità provvisoria rilevando errore nel file Configuration.Yaml.
--------------------------------------
Invalid config for [http]: not a file for dictionary value @ data['http']['ssl_certificate']. Got '/home/pi/homeassistant/dehydrated/certs/casa1.duckdns.org/fullchain.pem'
not a file for dictionary value @ data['http']['ssl_key']. Got '/home/pi/homeassistant/dehydrated/certs/casa1.duckdns.org/privkey.pem'. (See /config/configuration.yaml, line 18). 
-------estratto file configuration.Yaml------------------
14  duckdns:
15    domain: casa1
16    access_token: 123456789123456789123456789
17
18 http:  
19   ssl_certificate: /home/pi/homeassistant/dehydrated/certs/casa1.duckdns.org/fullchain.pem
20   ssl_key: /home/pi/homeassistant/dehydrated/certs/casa1.duckdns.org/privkey.pem 

ho controllato i file richiamati, e mi sembrano scritti correttamente. 
potete darmi un consiglio ?
grazie

**Comment by gforment** [comment]
probabilmente il path indicato è quello del sistema operativo, docker solitamente lo vede sotto /config. Se usi docker, i path devono essere del container, prova a controllare

**Reply by snakuzzo** [reply]
Come ti ha detto @gforment, all'interno del container hai altra struttura directory e probabilmente quel percorso di dehydrated dove avevi i certificati, non è stato mappato nel container.
Ti faccio un esempio...
ammettiamo tu abbia:

config dir HA `/home/pi/ha_config_dir`
dehydrated dir -> `/home/pi/dehydrated`

Configuri il container per mappare `/home/pi/ha_config_dir` sotto `/config`

```
-v /home/pi/ha_config_dir:/config
```

Home Assistant girerà all'interno del container Docker che avrà la sua struttura di drectory e sotto la sua `/config` vedrà quanto presente sotto `/home/pi/ha_config_dir`
La directory `/home/pi/dehydrated`, non essendo stata mappata, non potrà mai essere vista.
Per ovviare dovresti ricreare il container aggiungendo un nuovo volume per l'altra directory

```
-v /home/pi/ha_config_dir:/config
-v /home/pi/dehydrated:/dehydrated
```

E dovrai quindi adeguare il configuration.yaml cambiando il path dei certificati con quello che lui vedrà all'interno del container...
Ad esempio...

```
ssl_certificate: /dehydrated/certs/casa1.duckdns.org/fullchain.pem
ssl_key: /dehydrated/certs/casa1.duckdns.org/privkey.pem 
```

Detto ciò, ti consiglio vivamente di lasciar perdere dehydrated e di fare in maniera diversa utilizzando Caddy come reverse proxy che si preoccuperà per te di gestire i certificati (Sul sito trovi la guida)

Prima di tutto però ti consiglierei di studiare un minimo Docker e di fare alcune prove per capire bene soprattutto questa cosa dei volumi e su come funzionano le network. Docker è comodo, semplice, ma prima di utilizzarlo tocca capirlo, altrimenti si va avanti a botta di copia-incolla senza capire...e questo è male

**Reply by indomus** [reply]
Se stai migrando, [segui questa guida](https://indomus.it/guide/come-migrare-home-assistant-core-da-ambiente-venv-a-docker-su-raspberry-pi/) e, come detto da Diego, usa Caddy

**Reply by gericom90** [reply]
Ok, grazie mille

**Comment by indomus** [comment]
che c'entra dehydrated?

**Reply by gericom90** [reply]
stavo seguendo una guida (magari vecchiotta), ma ora su consiglio di snakuzzo, vedro di installare Caddy.
grazie ancora
